### PR TITLE
Feature/tao 7617/persist highlighted text

### PR DESCRIPTION
**Pull request by oatymart**
It's a small part of https://oat-sa.atlassian.net/browse/TAO-7617

The main change is to enable the list of blacklisted containers to be extended with a parameter passed to the highlighter. A change to the logic was necessary because the blacklisted containers had not been properly implemented.

I've added a test to cover a passed-in blacklist parameter.

### CODE REVIEW

**Comment by krampstudio** [comment]
Could you please describe the changes you did ?
Are they covered by the tests ?

File: views/js/ui/highlighter.js [hunk 1]
```diff
@@ -36,7 +36,7 @@ define([
      * Children of those nodes types cannot be highlighted
      * @type {string[]}
      */
-    var containersBlackList = [
+    var _defaultBlackList = [
```
Comment (krampstudio) [inline views/js/ui/highlighter.js]: Even if you saw it somewhere, we had a discussion regarding this convention (to prefix private variable by an underscore) and we came up to the conclusion that's not what we wanted. Variable names should be explicit enough to not used this kind of convention (and here the name you've selected is sufficient : `defaultBackList`)
Could you please unprefix your variable.

File: views/js/ui/highlighter.js [hunk 2]
```diff
@@ -47,11 +47,21 @@ define([
      * @param {Object} options
      * @param {Object} options.className - name of the class that will be used by the wrappers tags to highlight text
      * @param {Object} options.containerSelector - allows to select the root Node in which highlighting is allowed
+     * @param {Object} [options.containersBlackList] - additional blacklist selectors to be added to module instance's blacklist
+     * @param {String} [options.id] - key for identification/storage
```
Comment (krampstudio) [inline views/js/ui/highlighter.js]: This entry has been added by mistake ?

File: views/js/ui/highlighter.js (hunk 1 above)
Comment (oatymart) [reply views/js/ui/highlighter.js]: ok, done.

File: views/js/ui/highlighter.js (hunk 2 above)
Comment (oatymart) [reply views/js/ui/highlighter.js]: Was used for debugging, I've removed it.

**Pull request merged**